docs/DeveloperGuide.md

```diff
@@ -151,7 +151,7 @@ The `Model` component,
 
 <box type="info" seamless>
 
-**Note:** The module plan data is split into different semesters (e.g. Year 1 S1, Year 1 S2, Year 2 S1, etc). Instead of one `UniqueModuleList` storing all of the User's modules across multiple semesters, each semester's modules are stored in their own `UniqueModuleList` object. Nevertheless, modules are required to be unique across semesters, meaning that the same module will be prevented from being added to multiple semesters.
+**Note:** The module plan data is split into different semesters (e.g. Year 1 S1, Year 1 S2, Year 2 S1, etc). Instead of one `UniqueModuleList` storing all of the user's modules across multiple semesters, each semester's modules are stored in their own `UniqueModuleList` object. Nevertheless, modules are required to be unique across semesters, meaning that the same module will be prevented from being added to multiple semesters. The implementation of this checked can be found in `ModelManager`.
```

Comment:
I think `ModulePlanSemesterList` or `ModulePlan` might shed more light on the implemetation for this explanation. Perhaps it can also link to the file itself in github.